sdk/core/tests/vmquery_test.rs

```diff
@@ -0,0 +1,612 @@
+multiversx_sc::imports!();
+multiversx_sc::derive_imports!();
+
+use multiversx_sc::types::{BigUint, ManagedAddress, ManagedBuffer};
+use multiversx_sc_scenario::DebugApi;
+use multiversx_sdk::data::address::Address;
+use multiversx_sdk::data::types::native::{NativeValue, NativeValueManagedVecItem};
+use multiversx_sdk::data::types::payment::Payment;
+use multiversx_sdk::data::vm::{VMOutputApi, VmValueRequest};
+
+const CONTRACT_ADDRESS: &str = "erd1qqqqqqqqqqqqqpgq7ykazrzd905zvnlr88dpfw06677lxe9w0n4suz00uh";
+const CALLER_ADDRESS: &str = "erd1devnet6uy8xjusvusfy3q83qadfhwrtty5fwa8ceh9cl60q2p6ysra7aaa";
+const FUNC_NAME: &str = "";
+const VALUE: &str = "0";
+
+const SAMPLE_RETURN_DATA: [&'static str; 8] = [
+    "VGhpcyBpcyBhIGJ1ZmZlcg==", // "This is a buffer",
+    "DeC2s6dkAAA=", // 1e18
+    "AAAAAAAAAAAFAPEt0QxNK+gmT+M52hS5+te982SufOs=", // erd1qqqqqqqqqqqqqpgq7ykazrzd905zvnlr88dpfw06677lxe9w0n4suz00uh,
+    "AAAAFFRoaXMgaXMgYSBzYW1wbGUgc3RyAAAAAgPoAAAAAAAAAAAFAPEt0QxNK+gmT+M52hS5+te982SufOs=", // SampleCodable : "This is a sample str", 1000, erd1qqqqqqqqqqqqqpgq7ykazrzd905zvnlr88dpfw06677lxe9w0n4suz00uh
+    "V0VHTEQtYWJjZGVm", // "WEGLD-abcdef"
+    "Zmlyc3Q=", // "first"
+    "c2Vjb25k", // "second"
+    "dGhpcmQ=" // "third"
+];
```

Comment:
You should take a look at interactors/snippets (preferably in rc/v0.42). We have a full framework for querying the blockchain with SC abstractions and automated data parsing, plus ability to automatically generate integration tests from the interactions.

Reply:
I didn't know! I checked it and this is really great! I'll wait for when you merge it into main before continuing with what I want to achieve